src/dlgPackageExporter.cpp

```diff
@@ -145,26 +146,26 @@ void dlgPackageExporter::recurseTree(QTreeWidgetItem* pItem, QList<QTreeWidgetIt
     }
 }
 
-bool dlgPackageExporter::writeFileToZip(const QString& archiveFileName, const QString& fileSystemFileName, zip* archive)
+std::pair<bool, QString> dlgPackageExporter::writeFileToZip(const QString& archiveFileName, const QString& fileSystemFileName, zip* archive)
 {
     struct zip_source* s = zip_source_file(archive, fileSystemFileName.toUtf8().constData(), 0, -1);
     if (s == nullptr) {
-        displayResultMessage(tr("Failed to open file \"%1\" to place into package. Error message was: \"%2\".",
-                                // Intentional comment to separate arguments
-                                "This error message will appear when a file is to be placed into the package but the code cannot open it.")
-                             .arg(fileSystemFileName, zip_strerror(archive)), false);
-        return false;
+        return {false,
+                tr("Failed to open file \"%1\" to place into package. Error message was: \"%2\".",
+                   // Intentional comment to separate arguments
+                   "This error message will appear when a file is to be placed into the package but the code cannot open it.")
+                        .arg(fileSystemFileName, zip_strerror(archive))};
     }
 
-    if (zip_file_add(archive, archiveFileName.toUtf8().constData(), s, ZIP_FL_ENC_UTF_8|ZIP_FL_OVERWRITE) == -1) {
-        displayResultMessage(tr("Failed to add file \"%1\" to package \"%2\". Error message was: \"%3\".",
-                                // Intentional comment to separate arguments
-                                "This error message will appear when a file is to be placed into the package but cannot be done for some reason.")
-                             .arg(archiveFileName, mPackagePathFileName, zip_strerror(archive)), false);
-        return false;
+    if (zip_file_add(archive, archiveFileName.toUtf8().constData(), s, ZIP_FL_ENC_UTF_8 | ZIP_FL_OVERWRITE) == -1) {
+        return {false,
+                tr("Failed to add file \"%1\" to package. Error message was: \"%3\".",
+                   // Intentional comment to separate arguments
+                   "This error message will appear when a file is to be placed into the package but cannot be done for some reason.")
+                        .arg(archiveFileName, zip_strerror(archive))};
```

Comment:
Still some deallocation needed:
```suggestion
        zip_source_free(s);
        s = nullptr;
        return {false,
                tr("Failed to add file \"%1\" to package. Error message was: \"%3\".",
                   // Intentional comment to separate arguments
                   "This error message will appear when a file is to be placed into the package but cannot be done for some reason.")
                        .arg(archiveFileName, zip_strerror(archive))};
```